controllers/configurationpolicy_controller.go

```diff
@@ -1307,17 +1314,37 @@ func (r *ConfigurationPolicyReconciler) determineDesiredObjects(
 		return nil, nil, errEvent, err
 	}
 
-	relevantNamespaces := []string{}
+	// Set up relevant object namespace-names map to populate with the
+	// namespaceSelector and objectSelector
+	relevantNsNames := map[string][]string{}
 
-	// Fetch and filter namespaces using provided namespaceSelector
 	desiredNs := parsedMinMetadata.Metadata.Namespace
 
 	// If the namespace is templated, consider it not explicitly set
-	if templates.HasTemplate([]byte(parsedMinMetadata.Metadata.Namespace), "", true) {
+	if templates.HasTemplate([]byte(desiredNs), "", true) {
 		desiredNs = ""
 	}
 
-	// The object is namespaced and either has no namespace specified or it is templated in the object definition
+	desiredName := parsedMinMetadata.Metadata.Name
+
+	// If the name is templated, consider it not explicitly set
+	if templates.HasTemplate([]byte(desiredName), "", true) {
+		desiredName = ""
+	}
+
+	// Set up default name array to be used for each namespace. If no
+	// objectSelector is provided, add the desired name as the default.
+	defaultNamesPerNs := []string{}
+	objectSelector := objectT.ObjectSelector
+	fetchNames := desiredName == "" && objectSelector != nil
```

Comment:
Is this just boolean? Can we rename this? fetchNamea look like string array

Reply:
Good point. I was originally thinking I'd need it more than one place, but it looks like I don't. I'll just move it into the conditional directly.